ena-submission/scripts/submission_db_helper.py

```diff
@@ -122,10 +135,14 @@ def find_conditions_in_db(db_conn_pool, table_name, conditions):
     con = db_conn_pool.getconn()
     try:
         with con, con.cursor(cursor_factory=RealDictCursor) as cur:
-            query = f"SELECT * FROM {table_name} WHERE "
-            query += " AND ".join([f"{key}='{str(value)}'" for key, value in conditions.items()])
+            # Prevent sql-injection with table_name validation
+            TableName.validate(table_name)
+            query = f"SELECT * FROM {table_name}"
+
+            where_clause = " AND ".join([f"{key}=%s" for key in conditions])
```

Comment:
I think that this implementation is already entirely safe for now, but to make the function even more future-proof, it would be good to validate the keys as well. A simple implementation, where you wouldn't need to list all possible keys, and that should be enough for this case, could be to check that the key only consists of letters (A-Za-z), numbers, and underscores.

Reply:
I added a function to perform an explicit check that all keys in the conditions dictionary are actual columns.